app/models/medium.rb

```diff
@@ -240,10 +240,8 @@ def self.search_sorts(search_params)
   # by the controller
   def self.search_by(search_params, page)
     search_params[:types] = [] if search_params[:all_types] == '1'
-    if search_params[:teachable_inheritance] == '1'
-      search_params[:teachable_ids] = Course.search_inherited_teachables(search_params)
-    end
-    search_params[:teachable_ids] = [] if search_params[:all_teachables] == '1'
+    search_params[:teachable_ids] = TeachableParser.new(search_params)
+                                                   .teachables_as_strings
```

Comment:
This line is not covered by tests explicitly.

Reply:
I would like to postpone that until I deal with tests for `medium.rb`. I will need to think about how to test this Solr stuff in `medium#search_by` best, and there will be probably a bigger setup needed for testing this method, so I would prefer not to write anything ad hoc right now as it would most likely have to be changed a lot later anyway.